fastfield_codecs/src/lib.rs

```diff
@@ -40,13 +40,15 @@ pub trait FastFieldCodecSerializer {
     /// Serializes the data using the serializer into write.
     /// There are multiple iterators, in case the codec needs to read the data multiple times.
     /// The iterators should be preferred over using fastfield_accessor for performance reasons.
-    fn serialize(
-        write: &mut impl Write,
-        fastfield_accessor: &impl FastFieldDataAccess,
+    fn serialize<W>(
+        write: &mut W,
```

Comment:
how is it different?

Reply:
Initially I wanted to reduce the number of generic parameters by moving the two `impl Iterator` into one generic, but that wasn't possible, because two closures are always different types, even when identical. The `W` is a residue of that refactor I forgot to rollback.